README.md

```diff
@@ -1,32 +1,151 @@
+1. Clone this repository to your local environment.
+2. Run `npm install` to install all the necessary packages required.
+3. Open `examples/index.html` in your browser to look at the preview of the library.
+
+### _Instructions for a developer_
+1. Install grunt - https://gruntjs.com/installing-grunt.
+2. Make the changes you are working on in respective /src files.
+3. Run `grunt build` to generate files in the /dist directory.
+4. Run `grunt transpile` to transpile es6 code and copy files needed to run the tests to the /dist directory.
+5. Run `grunt jasmine` to run tests on the LEL layers and ensure they pass.
+6. Test your changes on a browser by opening `examples/index.html`.
+
+## Features
+
+### _Zoom or Pan_
+
+Click and drag the map to pan it.
 
-The library is integrated to following places
+### _Change the Base Map and Overlay layers_
 
-* https://mapknitter.org/
-* https://publiclab.org/maps/
+Use the button on right-most corner to change the way the background of the map looks.
 
-## About
+### _See More Data_
 
-The information of each layer can be found here:
+- Toggle certain layers on and off using the Layers button in the toolbar. 
+- Layers with near-real-time or real-time data will have the 'NRT/RT' mark on them.
+- More information on the layer data will be available when clicking the 'i' button on the layer
+- Layers that allow contributions will have a `report` button or `contribute` button.
+- Layers will be visible on the menu only when the map view intersects with the layer's bounds or zoom levels.
+- A badge displays the number of new layers in the map view when the map intersects with new layers
 
-https://publiclab.org/notes/sagarpreet/06-06-2018/leaflet-environmental-layer-library
+### _Click on a Point_
+
+Click on a point or marker on the map to learn more about it.
+
+### _Minimal mode_
+
+Click on the button group on the left, below the zoom controls, to change between default markers mode and minimal markers mode. Use minimal markers mode for a smoother experience when using multiple layers with many markers.
+
+### _URL Hash_
+
+The map page's URL hash updates on map movement and when a layer is added or removed from a map. This helps preserve map state when refreshing or copying the URL to another page.
+
+### _Embed Code_
+
+Click on the button at the bottom on the left side of a map to generate an embed code so that the map page can be embedded in other sites.
+
+## Layers
+The information of each layer can be found here: [Layer Information](https://publiclab.org/notes/sagarpreet/06-06-2018/leaflet-environmental-layer-library)
 
-### Layers
 
 | Layer Name                  | Color   |
 | --------------------------- | ------- |
 | wisconsin                   | N/A     |
 | fracTrackerMobile           | N/A     |
 | purpleLayer                 | #8b0000 |
 | purpleairmarker             | #800080 |
-| skyTruth                    | #ff0000 |
+| skytruth                    | #ff0000 |
```

Comment:
Had somehow missed changing this last time. Fixed it to be consistent with the layer names used in the hash.

Reply:
So, one thing here could be linking to the info.json file for the layer, which would include the description text that gives the layer more context. That way also we wouldn't have to write out such text 2x -- what do you think?

Reply:
Oh I like that idea. Link right to the lines with the info, that way we don't have to keep updating the readme.

Reply:
I think that's a great idea! 😃

Reply:
So, let me get this straight. We would be linking each layer name here to its information in the info.json file?

Reply:
That's what I would do, yes! :smiley:

Reply:
We do not have all the layers listed in info.json whereas layersData.json has information for all layers. Because there is a lot of overlap between the two layers I am thinking it is best to combine them than repeat information for all layers again in the info.json file and also that it is best to do it now than later. I recall there were concerns regarding the overlap in the information before. I will attempt this in a separate PR as I may have to refactor some code in the layersBrowser.js util. Once that is done I'll link the layer names here to their respective section in the info.json file.

Reply:
Sounds great. Thank you!